apps/prairielearn/elements/pl-file-editor/pl-file-editor.py

```diff
@@ -20,6 +20,35 @@
 DIRECTORY_DEFAULT = "."
 NORMALIZE_TO_ASCII_DEFAULT = False
 
+range_default = "false" #flag for editor
+
+def categorize_options(element, data):
+    file_contents = ""
+    range_list = []
+    line = 0 # keeps track of what line in the editor we are in
+
+    for child in element:
+        child_html = pl.inner_html(child)
+        if child_html[-1] != "\n":
+            child_html += "\n"
+        file_contents += child_html
```

Comment:
Strings are immutable in Python, so using `+=` creates a new string each time it's called. You should put all of the strings you wish to concatenate in a list and then call `"".join(list)` instead.

Reply:
I addressed this issue in the latest commit, thanks for the feedback!